solc/jsonCompiler.cpp

```diff
@@ -68,13 +68,13 @@ ReadFile::Callback wrapReadCallback(CStyleReadFileCallback _readCallback = nullp
 				result.success = false;
 				result.contentsOrErrorMessage = "File not found.";
 			}
-			if (contents_c)
+			else if (contents_c)
```

Comment:
I think these two changes here should be reverted. If `error_c` is set, then it should be treated as an error. Furthermore, if both are set, then both should at least be `free`d.